### ISSUE TITLE
BUGFIX - gold+iron smelters sound problem

### ISSUE THREAD

**Issue by p1plp1** [freeserf/freeserf#478]
DataSourceDOS::get_sound(size_t index) { needs to be bugfixed for sound with if (index == 62),
when this resource is played it bubles only once and then high tone beeeep noise only ... any idea how to load this resource and cut only real data and play it correctly?

**Comment by sternenzaehler** [comment]
I will try to look into it.

**Comment by tlongstretch** [comment]
I had tried applying p1plp1's personal fix in my fork but it didn't seem to work so I reverted it.

**Comment by p1plp1** [comment]
that was not fix :) that was only problem finding, for iron smelter sound with index 62 is there still problem, i tried to cut off and force sound lenght for index 62 with some constant value but without success, any progress anyone?

**Comment by tlongstretch** [comment]
My work-around is to play the Amiga sound instead.  This could take a ton of time to solve the root cause, it doesn't seem worth the effort to me

**Comment by p1plp1** [comment]
hello tlongstretch can i ask u for actual bugfixed whole compiled +sources packed into zip? i have difficulties to compile and run ... :(

**Comment by tlongstretch** [comment]
Of Forkserf or Freeserf?  On Forkserf "Releases" area I have pre-compiled binaries for linux and windows, though I've only tested the legacy glibc versions personally.  I can probably compile Freeserf also if you're having issues compiling it, though I thought it was already published here on Freeserf project.  

https://github.com/forkserf/forkserf/releases

**Comment by tlongstretch** [comment]
When you say Amiga color palette is bad, what are you expecting it to look like?   I had always assumed this is what it originally looked like as I never saw the Amiga version.  Now, looking at screenshots if I google for "Settlers Amiga" it looks much like the DOS version.  I think you are correct that the terrain is corrupted, but I'm not sure if there's anything that can be done to fix it.  It looks same to me in Freeserf or Forkserf
![Screenshot from 2023-01-04 22-12-56](https://user-images.githubusercontent.com/45954803/210693477-79896bc7-4af6-4123-96c3-7910afb23e60.png)

**Comment by tlongstretch** [comment]
Regarding the DOS boils sound, I tried experimenting with editing it chunk by chunk as it loads and no success.

**Comment by jacefuse** [comment]
Unless I am mistaken, the Amiga version of Settlers was using Extra Half Bright. The reason the colors may come out wrong is probably because the bitmap uses 64 colors but only 32 are defined in the pallet. The upper 32 colors are a copy of the lower 32 colors but only half as bright. If there are colors defined in the upper 32 colors of the pallet, they may not be the correct colors.

--- END OF MESSAGE! ---

**Comment by p1plp1** [comment]
yes, looks like that mess on that picture, pallete must be fixed prior of displaying any sprites etc.

**Comment by tlongstretch** [comment]
you might be correct about the fix.   Is there a significant different in the (correct) Amiga graphics versus DOS?  The non-terrain sprites look extremely similar, and the DOS ones look better.  The only ones where Amiga looks arguably better are the Castle and Garrison sprites, where Amiga has a brown roof while DOS has red.  

![amiga_vs_dos_buildings](https://user-images.githubusercontent.com/45954803/210886363-a0e377a0-4245-40cf-bffd-1634356c76af.png)

**Comment by GraionDilach** [comment]
The star overlay on the ingame bottom bar's buttons when they are active is only animated in Amiga AFAIK.

**Comment by tlongstretch** [comment]
You are right, that spinning star does look slick.  I just watched some youtube amiga settlers gameplay.  I don't see anything else that looks significant, though. I think implementing the amiga spinning star and otherwise using DOS graphics is the way to go!

**Comment by tlongstretch** [comment]
it seems Freeserf does not implement the spinning star even if using Amiga graphics, unfortunately

**Comment by p1plp1** [comment]
about boiling sound - i have just unconfirmed idea that problem can be in sound extraction algorythm, maybe i try to expand data and save it as playable wav sound and test it in audio player, if problem happends just during loading sound from file due to wrong algorythm.

**Comment by p1plp1** [comment]
about castle, storage - hmmm that brown roof + rock wall looks better but if u look at road at front of the storage it looks dirty in amiga version, that red / brown roof is maybe rooot of that pallete problem --> what about shifting brown to be more red? maybe it fixes that terrain desert textures too ...

**Comment by p1plp1** [comment]
now im sure i have problems with mingw64 compiler and threads on Windows x64 ... dunno exactly why but i get SDL_CreateMutex errror hm, i updated mingw during time ...

**Comment by tlongstretch** [comment]
maybe try VSCode?  That is what I am using on linux, and I was using VisualStudio Community Edition when I was developing on Windows

**Comment by tlongstretch** [comment]
I got the amiga spinning star working as a prototype.  I had to extract the spinning star frame by frame from a youtube video, if anybody knows how to extract it from the amiga data files that would be better, but this still looks pretty good 

https://user-images.githubusercontent.com/45954803/213021767-ae0a541b-84f6-4868-b41c-dfe98df0c532.mp4

**Comment by zdechlak** [comment]
https://thecompany.pl/gra/Settlers%7C+The

here you can play settlers via amiga emulator
